packages/server/src/adapters/content-handlers/selectContentHandlerOrUnsupportedMediaType.ts

```diff
@@ -1,20 +1,35 @@
 import { TRPCError } from '../../@trpc/server';
-import type { BaseContentTypeHandler } from '../../@trpc/server/http';
+import type {
+  BaseContentTypeHandler,
+  HTTPHeaders,
+} from '../../@trpc/server/http';
+
+interface MinimalHandlerOpts {
+  req: {
+    headers: HTTPHeaders | Headers;
+  };
+}
 
 export function selectContentHandlerOrUnsupportedMediaType<
-  THandlerOpts,
+  THandlerOpts extends MinimalHandlerOpts,
   THandler extends BaseContentTypeHandler<THandlerOpts>,
 >(handlers: THandler[], opts: THandlerOpts) {
-  const handler = handlers.find((handler) => handler.isMatch(opts));
+  const contentType = new Headers(opts.req.headers as HeadersInit).get(
+    'content-type',
+  );
+
+  const error = new TRPCError({
```

Comment:
Maybe we should still only create this when we actually need it? Unnecessary to create it for every request